packages/x-data-grid/src/components/menu/GridMenu.tsx

```diff
@@ -113,8 +135,10 @@ function GridMenu(props: GridMenuProps) {
       as={rootProps.slots.basePopper}
       className={clsx(className, classes.root)}
       ownerState={rootProps}
-      open={open}
-      anchorEl={target as any}
+      open={openProp}
+      anchorEl={
+        useLazyRef(() => getTarget(target, targetRef)!).current
+      }
```

Comment:
Not sure what to do. I've been juggling between fixing the issue, working around this [popper bug](https://github.com/mui/mui-x/issues/12952), and not breaking the public API (so `target` type must stay backwards compatible).

Maybe I should just ignore the popper bug and we consider that an upstream issue? The popper bug only appears if we have a menu opened at mount time (before the refs are set, so the menu is positionned at `0,0`). I don't think it's something that realistically happens to our users, but we have an argos test that matches that case, thus why I've been trying to fix it with the target ref hack. I could instead modify our test so that we open the menu in a timeout. That way we don't hit the popper bug. Wdyt?

Reply:
> I've been juggling between fixing the issue, working around this [popper bug](https://github.com/mui/mui-x/issues/12952), and not breaking the public API

Does the suggested code snippet break one of these things?

> I could instead modify our test so that we open the menu in a timeout.

This sounds good to me, I don't think we should spend more time just to fix the "menu opened at mount time" issue.

Reply:
If I don't have to fix the popper bug the whole target change can disappear.